service/service.go

```diff
@@ -198,23 +196,23 @@ func (srv *Service) Start(ctx context.Context) error {
 	)
 
 	// enable status reporting
-	srv.reporter.Ready()
+	srv.host.Reporter.Ready()
```

Comment:
I think we can remove `Ready()` and the ready related code in internal/status. When `ReportStatus` was on `TelemetrySettings` it was possible to call it before start. Now that it's been moved to `Host` this should no longer be possible. We can create an issue and address this in a followup. I'd be willing to take that task if we agree.

Reply:
I've captured this in https://github.com/open-telemetry/opentelemetry-collector/issues/10792.  This PR is already complicated enough I think that can be done in a followup.